tls/s2n_prf.c

```diff
@@ -408,6 +409,31 @@ int s2n_hybrid_prf_master_secret(struct s2n_connection *conn, struct s2n_blob *p
     return s2n_prf(conn, premaster_secret, &label, &client_random, &server_random, &conn->kex_params.client_key_exchange_message, &master_secret);
 }
 
+int s2n_prf_calculate_master_secret(struct s2n_connection *conn, struct s2n_blob *premaster_secret)
+{
+    POSIX_ENSURE_REF(conn);
+
+    if (conn->ems_negotiated && S2N_IN_TEST) {
+        uint8_t data[S2N_MAX_DIGEST_LEN] = { 0 };
+        struct s2n_blob digest = { 0 };
+        POSIX_GUARD(s2n_blob_init(&digest, data, sizeof(data)));
+
+        if (conn->mode == S2N_CLIENT) {
+            POSIX_GUARD(s2n_handshake_finish_header(&conn->handshake.io));
+        }
+        POSIX_GUARD(s2n_stuffer_reread(&conn->handshake.io));
+        struct s2n_blob client_key_message = { 0 };
```

Comment:
Does resetting the read pointer for handshake.io interfere with the caller (I assume a handler) reading from handshake.io? Can we be sure the caller is done reading when this executes?

Might be worth a fresh stuffer, like `struct s2n_stuffer message = conn->handshake.io`.